### PR TITLE
Adds support for iframe nodes

### PR DESCRIPTION
**Pull request by dwsmart**
Adds iframe support, extracts the child nodes from contentDocument

### CODE REVIEW

**Comment by dwsmart** [comment]
@googlebot I signed it!

**Comment by dwsmart** [comment]
Added parameter to make iframe content optional, defaulted to true, addresses #3

File: subtree.js [hunk 1]
```diff
@@ -43,9 +43,21 @@ module.exports = {
       return elem;
     }
 
+    // if it's an iframe, copy the children of contentDocument.
+    if(iframe) {
```
Comment (AVGP) [inline subtree.js]: Nitpick: As the only code that runs is the entirety of the if-clause below, I'd prepend this to the if condition in the next line.

File: serialize.js [hunk 2]
```diff
@@ -18,7 +18,7 @@ const { JSDOM } = jsdom; // ughhh, why?
 const domNode = require('./dom-node');
 const subtree = require('./subtree');
 
-module.exports = function serialize(domLike) {
+module.exports = function serialize(domLike, iframe = true) {
```
Comment (AVGP) [inline serialize.js]: I prefer when boolean parameters have names that designate that they're boolean, e.g. `includeIframes` or `inlineIframes` or something like that.

File: subtree.js [hunk 3]
```diff
@@ -23,7 +23,7 @@ module.exports = {
    * @param {function} nodeCallback (optional) A callback that receives the node-like and the created element after attributes are copied but before the subtree is created.
    * @returns {JSDOM.Node} The DOM subtree
    */
-  constructSubtreeForNode: function(document, nodeLike, nodeCallback) {
+  constructSubtreeForNode: function(document, nodeLike, iframe, nodeCallback) {
```
Comment (AVGP) [inline subtree.js]: See my comment on parameter names for boolean parameters in serialize.js - this should probably match

Comment (dwsmart) [reply subtree.js]: fixed in newest commit

File: subtree.js (hunk 1 above)
Comment (dwsmart) [reply subtree.js]: fixed in newest commit

**Pull request merged**